log.js

```diff
@@ -0,0 +1,106 @@
+const { writeFileSync, readFileSync } = require("fs");
+const { stringify, parse } = require("yaml");
+
+// log file
+const logFile = "log.yaml";
+
+// load previous runs of bot from log file
+function loadLog() {
+  try {
+    const file = readFileSync(logFile, { encoding: "utf8" });
+    const data = parse(file);
+    return data;
+  } catch (error) {
+    return null;
+  }
+}
+
+// save this run of bot to log file
+function saveLog({ preprint, comment, tweets }, log = []) {
+  try {
+    log.push({
+      preprint: cleanPreprint(preprint),
+      comment: cleanComment(comment),
+      tweets: cleanTweets(tweets),
+    });
+
+    writeFileSync(logFile, stringify(log), () => null, { encoding: "utf8" });
+    return true;
+  } catch (error) {
+    return null;
+  }
+}
+
+// keep only useful props from preprint data
+const cleanPreprint = (preprint) => {
+  for (const authorIndex of Object.keys(preprint.authors)) {
+    delete preprint?.authors[authorIndex]?.id;
+    delete preprint?.authors[authorIndex]?.institution;
+  }
+  delete preprint?.ranks;
+  delete preprint?.publication;
+
+  return preprint;
+};
+
+// keep only useful props from comment data
+const cleanComment = (comment) => {
+  delete comment?.editableUntil;
```

Comment:
Similar to below - a Q as to whether it's better to keep selected elements or remove the others.